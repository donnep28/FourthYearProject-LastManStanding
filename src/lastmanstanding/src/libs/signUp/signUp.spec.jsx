import React from 'react';
import {shallow} from 'enzyme';
import SignUp from './signUp';
import {fireEvent, render} from '@testing-library/react';

describe('joinLeagues', () => {
  beforeAll(() => {
    jest.resetAllMocks();
  });
  describe('Snapshot', () => {
    it('renders teams', () => {
      const component = shallow(
          <SignUp/>,
      );
      expect(component).toMatchSnapshot();
    });
  });
  describe('textFields', () => {
    it('updates username', () => {
      const {getByTestId} = render(<SignUp/>);
      const element = getByTestId('username');
      fireEvent.change(element, {target: {value: 'mockUsername'}});
      expect(element).toHaveValue('mockUsername');
    });
    it('updates email', () => {
      const {getByTestId} = render(<SignUp/>);
      const element = getByTestId('email');
      fireEvent.change(element, {target: {value: 'mockEmail'}});
      expect(element).toHaveValue('mockEmail');
    });
    it('updates password', () => {
      const {getByTestId} = render(<SignUp/>);
      const element = getByTestId('password');
      fireEvent.change(element, {target: {value: 'mockPassword'}});
      expect(element).toHaveValue('mockPassword');
    });
  });
});