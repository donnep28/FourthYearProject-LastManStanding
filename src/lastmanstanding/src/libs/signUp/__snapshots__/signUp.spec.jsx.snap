// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joinLeagues Snapshot renders teams 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <Memo(ForwardRef(LockOutlinedIcon)) />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Sign up
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      noValidate={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="uname"
            fullWidth={true}
            id="username"
            inputProps={
              Object {
                "data-testid": "username",
              }
            }
            label="Username"
            name="username"
            onChange={[Function]}
            required={true}
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="email"
            fullWidth={true}
            id="email"
            inputProps={
              Object {
                "data-testid": "email",
              }
            }
            label="Email Address"
            name="email"
            onChange={[Function]}
            required={true}
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <WithStyles(ForwardRef(TextField))
            autoComplete="current-password"
            fullWidth={true}
            id="password"
            inputProps={
              Object {
                "data-testid": "password",
              }
            }
            label="Password"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-4"
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        type="submit"
        variant="contained"
      >
        Sign Up
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Grid))
        container={true}
        justify="flex-end"
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Link))
            onClick={[Function]}
            style={
              Object {
                "color": "blue",
                "cursor": "pointer",
              }
            }
          >
            <p>
              Already have an account? Sign in
            </p>
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </div>
</WithStyles(ForwardRef(Container))>
`;
